### ISSUE TITLE
adding a gps module

### ISSUE THREAD

**Issue by runcom** [eried/FlipperZeroMayhem#41]
Hi, I'm trying to make a sense of how to add the gps module to the mayhem v2 board, is there any documentation about it? I can see the code mentioning the gps pins and such but wiring wise, I'm kind of lost. I can see that the mmc is set to use just 1 pin so I'm guessing the others should be free to wire up? or are they being used by the nrf/cc modules?

**Comment by eried** [comment]
Technically you only need 1 pin for gps, the RX. If you arent using the microsd, you could hack your way out using those pins OR [gpio16](https://lastminuteengineers.com/esp32-cam-pinout-reference/), but I think that is on the back. Either way I am not sure how easy this will be but report how it goes. 

BTW the gps part in the firmware is because this is a fork, so if you use a normal ESP32 dev board there is a lot of free pins

**Comment by runcom** [comment]
> Technically you only need 1 pin for gps, the RX. If you arent using the microsd, you could hack your way out using those pins OR [gpio16](https://lastminuteengineers.com/esp32-cam-pinout-reference/), but I think that is on the back. Either way I am not sure how easy this will be but report how it goes.

what I'm basically trying to do is to put esp32-cam + NRF24l01 / CC1101 + gps (according to [this wiring](https://github.com/justcallmekoko/ESP32Marauder/wiki/gps-modification#connections)) to allow wardriving marauder too with the flipper zero. I basically would love to have [this](https://www.tindie.com/products/eried/mayhem-v2-for-flipper-all-in-1-espcamsdnrfcc/) plus the GPS.

My understanding is that the fz is connected to the esp32-cam but also the nrf24 and cc1101 are connected to it - but the GPS module has to be attached to the esp32-cam.

I'll try with just one pin as you suggested but I want to use the microsd (and in 1bit mode, I should have a couple of gpios available right?)

**Comment by eried** [comment]
yeah, but I am not sure if they are exposed easily :S I dont remember in this moment if the gpio16 is actually in the back, I need to check later

**Comment by runcom** [comment]
I read everywhere that gpio16 is actually busy with psram tho - so it might be unusable :/  although I don't think i'll use the camera when doing wifi and gps - maybe trying pin 12 or 13 may be better if the microsd works with only one pin

**Comment by eried** [comment]
maybe, but I am not sure how easy is to use those pins, maybe you just need to test 1 by 1 which one is able to receive serial stream from the pin with SoftwareSerial(

**Comment by runcom** [comment]
This change worked for me:

```
     #elif defined(MARAUDER_FLIPPER)
       #define GPS_SERIAL_INDEX 1
-      #define GPS_TX 9
-      #define GPS_RX 21
+      #define GPS_TX 12
+      #define GPS_RX 13
```
with proper connection to the esp32-cam :) happy to send a PR too 

```
...
Satellites: 7
  Accuracy: 3.00
  Latitude: redacted
 Longitude: redacted
  Altitude: 422.80
  Datetime: 2024-6-2 17:53:58
....
```

**Comment by eried** [comment]
that would be great, specially adding the pinout :)
